File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/bookstock/stockBookChangeResponse.ts
import { BookResponse, mapBookResponseToBook, StockBookChange } from "../../../..";

export interface StockBookChangeResponse {
    id: number,
    bookId: number,
    book: BookResponse,
    changeAmount: number
}

export function mapStockBookChangeResponseToStockBookChange(response: StockBookChangeResponse): StockBookChange {
    return {
        id: response.id,
        bookId: response.bookId,
        book: mapBookResponseToBook(response.book),
        changeAmount: response.changeAmount,
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/StockBookChangeResponse.cs
using LibraryShopEntities.Domain.Dtos.Library;

namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class StockBookChangeResponse
    {
        public int Id { get; set; }
        public int BookId { get; set; }
        public BookResponse? Book { get; set; }
        public int ChangeAmount { get; set; }
    }
}
